No-vote column handling

karink520 opened the pull request:
1) Added functionality to handle passing a list `non_candidate_names` to several `RowByColumnEI` methods, so that so that various plots and summaries can show voting support calculated *out of* relevant voters and not just out of the entire population. 

For example, if one of the c columns in the model is `"abstain"` and the rest are candidates, and you'd like to summarize estimated support among the *non-abstaining* demographic groups, you could pass `non_candidate_names=["abstain"]` to the `summary` method.

The `non_candidate_names` argument can also be a list of length more than one, in case you want to exclude more than one column/voting outcome.

The following functions now accept an optional `non_candidate_names` argument:

`RowByColumnEI.summary`
`RowByColumnEI.plot`
`RowByColumnEI.plot_kdes`
`RowByColumnEI.plot_boxplots`
`RowByColumnEI.precinct_level_estimates`

The following methods already accepted an optional `non_candidate_names` argument

`RowByColumnEI.candidate_of_choice_report`
`RowByColumnEI.candidate_of_choice_polarization_report`

Several methods involve computations only for a particular, specified, candidate, so this features is not relevant for them. Those are:

`RowByColumnEI.polarization_report`
`RowByColumnEI.margin_report`

2) BUG FIX - fixed a bug where the districtwide preference samples for each group were being calculated based on precinct samples weighted by total precinct population, when in fact they should have been weighted by the precinct population *within each demographic group*!

gabeschoenbach reviewed (changes requested):
So good to have this functionality added in! I just have two very small questions/comments, otherwise very excited to merge

- Line 214 in `r_by_c.py`: to my eye this looks like it _does_ accept multiple non-candidate names, but there's a TODO to add that in. Is the TODO redundant, or am I missing something?
- Line 579 in `r_by_c.py`: "correspondes" -> "corresponds"

Otherwise looks great! and very good to separate RxC models and utils into new files/

karink520 commented:
Oh, awesome -- thanks for catching both of those. Now fixed. I also added a quick example of the non_candidate_names functionality to the santa clara example notebook.
